scripts/maintenance/migrate_project/src/db.py

```diff
@@ -128,22 +128,29 @@ def get_project_and_files_to_migrate(
 
     # check file_meta_data in the projects to migrate
     for project in projects_to_migrate:
-        user_id = project["prj_owner"]
         project_id = project["uuid"]
 
-        files_metadata_cursor = _get_file_meta_data_without_soft_links(
-            connection=src_conn, user_id=user_id, project_id=project_id
-        )
-        for result in files_metadata_cursor:
-            file_meta_data = dict(result.items())
-            file_uuid = file_meta_data["file_uuid"]
-
-            if _meta_data_exists_in_destination(dst_conn, file_uuid):
-                _red_message(f"SKIPPING, sync for {_file_summary(file_meta_data)}")
-                skipped_files_meta_data.append(file_meta_data)
-                continue
-
-            files_meta_data_to_migrate.append(file_meta_data)
+        # Since multiple users can generate files in the project
+        # and nodes can be deleted we copy over all the files that
+        # are available in the current node.
+        node_uuids = project["workbench"].keys()
+        already_processed_file: set[str] = set()
+        for node_uuid in node_uuids:
+            for result in _get_file_meta_data_without_soft_links(
+                connection=src_conn, node_uuid=node_uuid, project_id=project_id
+            ):
+                file_meta_data = dict(result.items())
+                object_name = file_meta_data["object_name"]
+                if object_name in already_processed_file:
+                    continue
+                already_processed_file.add(object_name)
+
+                if _meta_data_exists_in_destination(dst_conn, object_name):
```

Comment:
so if I get that right. in case a study or part of it already exists in the destination it will only skip these files?
Could we not check that beforehand? I.E. not copy anything in that case? I do not really see the point of copying parts of a study by default. The only use-case I see is when you repeatedly copy the same project and you know that the destination might already have some parts. Nevertheless I would not do that by default, it sounds dangerous.

Reply:
No, if anything prevents copying due to conflicts the script will gracefully abort with an error (line 164 `if len(skipped_files_meta_data) > 0 or len(skipped_projects) > 0:`) before mutating anything.

So your desired behavior is already in place :)

Reply:
perfect!